components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -1609,6 +1609,38 @@ public List<Resource> getResourcesByType(int tenantId, String resourceTypeId)
         }
     }
 
+    @Override
+    public void deleteResourcesByType(int tenantId, String resourceTypeId) throws ConfigurationManagementException {
+
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        try {
+            if (isMySQLDB()) {
+                /**
+                 * Need to first delete files since in MySql Clustering scripts for NDB engine `ON DELETE CASCADE`is no
+                 * longer supported for foreign keys on NDB tables when the child table contains a column that uses any
+                 * of the BLOB or TEXT types. Issue: (https://github.com/wso2/product-is/issues/12167).
+                 */
+                List<Resource> resourceList = getResourcesByType(tenantId, resourceTypeId);
+                if (resourceList.isEmpty()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("No resource found for the resourceTypeId: " + resourceTypeId);
+                    }
```

Comment:
Not a good practice to have a if block just for a log

Reply:
Can't we return the method?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5030/commits/89229d99304bf28cb30b59a43d181d33054a0be6